strawberryfields/apps/graph/utils.py

```diff
@@ -0,0 +1,143 @@
+# Copyright 2019 Xanadu Quantum Technologies Inc.
+r"""
+Graph functions
+===============
+
+**Module name:** :mod:`strawberryfields.future.apps.graph.utils`
+
+.. currentmodule:: strawberryfields.future.apps.graph.utils
+
+This module provides some ancillary functions for dealing with graphs. This includes
+:func:`is_undirected` to check if an input matrix corresponds to an undirected graph (i.e.,
+is symmetric) and :func:`subgraph_adjacency` to return the adjacency matrix of a subgraph when an
+input graph and subset of nodes is specified.
+
+Furthermore, the frontend :func:`~strawberryfields.future.apps.graph.dense_subgraph.find_dense`
+function allows users to input graphs both as a `NumPy <https://www.numpy.org/>`__ array
+containing the adjacency matrix and as a `NetworkX <https://networkx.github.io/>`__ ``Graph``
+object. The :func:`to_networkx_graph` function allows both inputs to be processed into a NetworkX
+Graph for ease of processing.
+
+Summary
+-------
+
+.. autosummary::
+    is_undirected
+    subgraph_adjacency
+    to_networkx_graph
+    is_subgraph
+    update_options
+
+Code details
+------------
+"""
+from typing import Union, Iterable
+
+import networkx as nx
+import numpy as np
+
+graph_type = Union[nx.Graph, np.ndarray]
+
+
+def to_networkx_graph(graph: graph_type) -> nx.Graph:
+    """Converts input into a NetworkX ``Graph`` (i.e., an undirected graph).
+
+    Given an input undirected graph of type ``graph_type = Union[nx.Graph, np.ndarray]``,
+    this function outputs a NetworkX graph of type ``nx.Graph``. The input ``np.ndarray`` must be a
+    matrix that is symmetric (i.e., satisfy :func:`is_undirected`) and also real.
+
+    Args:
+        graph (graph_type): input graph to be processed
+
+    Returns:
+        graph: the NetworkX graph corresponding to the input
```

Comment:
```suggestion
        nx.Graph: the NetworkX graph corresponding to the input
```

Reply:
(this just avoids ambiguity with any existing graph objects/classes/modules in the apps layer)

Reply:
Makes sense!